File: jvm/src/main/kotlin/org/faziodev/saywhatstarter/types/messages/GoogleBasicCardMessage.kt
package org.faziodev.saywhatstarter.types.messages

data class GoogleBasicCardMessage(val basicCard: GoogleBasicCard) : GooglePlatformMessage() {
    companion object {
        fun create(
            title: String = "",
            subtitle: String = "",
            formattedText: String = "",
            image: BasicCardImage? = null,
            button: BasicCardButton? = null
        ): GoogleBasicCardMessage = GoogleBasicCardMessage(GoogleBasicCard(
            title,
            subtitle,
            formattedText,
            image,
            if(button != null) listOf(button) else listOf()
        ))
    }
}

File: csharp/SayWhatStarterWebhook/Models/Messages/GoogleBasicCardMessage.cs
using System.Collections.Generic;

namespace SayWhatStarterWebhook.Models.Messages
{
    public class GoogleBasicCardMessage: GooglePlatformMessage
    {
        public GoogleBasicCard BasicCard { get; set; }

        public static GoogleBasicCardMessage Create(string title, string subtitle, string formattedText,
            BasicCardImage image, BasicCardButton button)
        {
            return new GoogleBasicCardMessage
            {
                BasicCard = new GoogleBasicCard
                {
                    Title = title,
                    Subtitle = subtitle,
                    FormattedText = formattedText,
                    Image = image,
                    Buttons = new List<BasicCardButton> {button}
                }
            };
        }
    }
}